src/dt2js.js

```diff
@@ -31,9 +70,9 @@ function getRAMLContext (ramlData) {
  * @param  {string} typeName - Name of the type to be converted.
  * @param  {conversionCallback} cb - Callback to be called with converted value.
  */
-function dt2js (ramlData, typeName, cb) {
+function dt2js (rootFileDir, ramlData, typeName, cb) {
```

Comment:
the issue with `rootFileDir` being a required argument of `dt2jsCLI()` is that it breaks backward compatibility:
 ```
$ cd examples/
$ node dt2js_example.js
ramldt2jsonschema/src/dt2js.js:81
    cb(new Error('Invalid RAML data'), null)
    ^

TypeError: cb is not a function
(...)
```
can/should we find a way to infer `rootFileDir` from where `dt2js` is being run from? I.e. `process.cwd()`

Reply:
The problem with using cwd, is that it will only allow for rational include paths if the initial raml file is in the current working directory itself. otherwise, the relative paths required would look incoherent (ie. a path of /test/examples/included_file.raml when the file is sitting right next to the initial raml file).

If this is an acceptable constraint, then we can use cwd.

Reply:
You're right. In that case I suggest to add this as a method. Method would be optional and default to `cwd`. This way it's backward compatible. Makes sense?

E.g.

```
var raml2json = require('ramldt2jsonschema')
var join = require('path').join
var fs = require('fs')

var basePath = '/path/to/base/path/'
var filePath = join(basePath, 'api.raml')
var ramlData = fs.readFileSync(filePath).toString()

raml2json.setBasePath(basePath)
raml2json.dt2js(ramlData, 'Cat', function (err, schema) {
  if (err) {
    console.log(err)
    return
  }
  console.log(JSON.stringify(schema, null, 2))
})
```

Reply:
got it.